pkg/controller/endpointmonitor/endpointmonitor_controller.go

```diff
@@ -117,10 +117,18 @@ func (r *ReconcileEndpointMonitor) Reconcile(request reconcile.Request) (reconci
 	delay := time.Until(createTime.Add(config.GetControllerConfig().CreationDelay))
 
 	for index := 0; index < len(r.monitorServices); index++ {
-		monitor := findMonitorByName(r.monitorServices[index], monitorName)
+		monitor, err := findMonitorByName(r.monitorServices[index], monitorName)
+		// if there was some error while getting a monitor, re-queue it to try on the next reconcile iteration
+		if err != nil {
+			return reconcile.Result{RequeueAfter: RequeueTime}, err
+		}
```

Comment:
If it's unable to create a monitor this will be an endless re-conciliation loop !

Reply:
But if it was only the API call error? E.g. API has returned 500 code, and we would be able to add this monitor on the next iteration?